src/writeData/subscriptions/context/subscription.list.tsx

```diff
@@ -30,19 +37,134 @@ export const DEFAULT_CONTEXT: SubscriptionListContextType = {
   loading: RemoteDataState.NotStarted,
   change: (_id: string) => {},
   currentID: null,
+  bulletins: {},
 } as SubscriptionListContextType
 
 export const SubscriptionListContext = React.createContext<
   SubscriptionListContextType
 >(DEFAULT_CONTEXT)
 
+export interface Bulletin {
+  timestamp: number
+  message: string
+}
+
+interface BulletinsMap {
+  [key: string]: Bulletin[]
+}
+
+const sanitizeBulletin = (bulletin: string) => {
+  const patterns = [
+    'Connection to (.*) lost \\(or was never connected\\) and connection failed.',
+  ]
+  const regexObj = new RegExp(patterns.join('|'), 'i')
+  const matched = bulletin.match(regexObj)
+
+  if (!!matched?.length) {
+    bulletin = matched[0]
+  }
+
+  return bulletin
+}
+
+const getBulletinsFromStatus = status => {
+  if (!status?.processors?.length) {
+    return []
+  }
+
+  const uniqueBulletins = status.processors
+    .filter(pb => !!pb.bulletins.length)
+    .map(pb => pb.bulletins)
+    .flat()
+    .map(pb => {
+      const message = sanitizeBulletin(pb.bulletin.message)
+      const timestamp = parseDate(pb.bulletin.timestamp)
+
+      return {
+        timestamp,
+        message,
+      }
+    })
+    .reduce((prev, curr) => {
+      const existing = prev?.[curr.message] ?? 0
+      prev[curr.message] = existing < curr.timestamp ? curr.timestamp : existing
+      return prev
+    }, {})
+  return Object.keys(uniqueBulletins).map(b => {
+    return {
+      message: b,
+      timestamp: uniqueBulletins[b],
+    } as Bulletin
+  })
+}
+
+const parseDate = (timeString: string) => {
```

Comment:
what exactly is this function doing?

Reply:
This is to parse the passed timestamp into a date so that we can do comparison for only getting the latest duplicate message.